### PR TITLE
GAN Models

### PR DESCRIPTION
**Pull request by UTSAVS26**
Closes: #22 

- **Title**: Added Various GAN Architectures for Medical Imaging
- **Name:** Utsav Singhal
- **Identify yourself:** SSOC'24 Contributor

### Describe the add-ons or changes you've made 📃

Implemented the following GAN architectures for medical imaging:
- BasicGAN
- ACGAN
- cGAN
- EBGAN
- InfoGAN

### Type of change ☑️

- [x] New feature (non-breaking change which adds functionality)

### How Has This Been Tested? ⚙️

Tested locally to ensure each GAN model functions correctly and generates synthetic medical images as expected. Unit tests for Basic GAN are in place and runnable with `python -m unittest test_BasicGAN.py`.

### Checklist: ☑️

- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/AI-Code/blob/main/README.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/AI-Code/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Note to reviewers 📄

Please review the added GAN models for correctness and adherence to project standards. Thank you!

### CODE REVIEW

File: GAN/Algorithms/README.md
Comment (Avdhesh-Varshney) [inline GAN/Algorithms/README.md:1]: not required.

File: GAN/Algorithms/requirements.txt
Comment (Avdhesh-Varshney) [inline GAN/Algorithms/requirements.txt:1]: not required.

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
@UTSAVS26 enlist these algorithms in readme file of `GAN`

**Comment by UTSAVS26** [comment]
> @UTSAVS26 enlist these algorithms in readme file of `GAN`

All the algorithms as been listed in the readme file of `GAN` and successfully removed unrequired files as mentioned.

**Review by Avdhesh-Varshney — APPROVED** [review]

**Pull request merged**